### PR TITLE
Add qubes version to boot menu

### LINKED ISSUE #4663: Add qubes version to boot menu (grub)
Make it easier to distinguish different installations (for example USB sticks).

### PR DESCRIPTION
**Pull request by alimirjamali**
fixes: https://github.com/QubesOS/qubes-issues/issues/4663

### CODE REVIEW

**Comment by alimirjamali** [comment]
@marmarek Archlinux build [failure](https://gitlab.com/QubesOS/qubes-linux-utils/-/jobs/7348314695) is because of this?

https://github.com/QubesOS/qubes-issues/issues/9220

**Comment by marmarek** [comment]
> @marmarek Archlinux build [failure](https://gitlab.com/QubesOS/qubes-linux-utils/-/jobs/7348314695) is because of this?

I don't think so. The `error: GPGME error: Inappropriate ioctl for device` error is something else and I think this is what later is seen as package signature verification failure.

File: grub/grub.qubes-kernel-vm-support [hunk 1]
```diff
@@ -6,3 +6,5 @@ if [ -r /usr/share/qubes/marker-vm ] &&
     GRUB_CMDLINE_LINUX="$GRUB_CMDLINE_LINUX xen_scrub_pages=0"
 fi
 GRUB_ENABLE_BLSCFG=false
+# Add qubes version to boot menu
+GRUB_DISTRIBUTOR="$(cat /etc/system-release)"
```
Comment (marmarek) [inline grub/grub.qubes-kernel-vm-support]: Default grub config tries to keep it short, IMO even with version added, it still should be short - without `release`, and not including version twice (as it is in system-release file)

**Review by marmarek — CHANGES_REQUESTED** [review]

File: grub/grub.qubes-kernel-vm-support (hunk 1 above)
Comment (alimirjamali) [reply grub/grub.qubes-kernel-vm-support]: I thought the same. Which one is more interesting? `Qubes 4.2.2` or `Qubes (R4.2)`?
From technical point of view, the former is better. But the latter might be more interesting.
or maybe maybe make it like `Qubes (R4.2.2)`? It is necessary to decide if the minor version is useful or not.

Comment (marmarek) [reply grub/grub.qubes-kernel-vm-support]: I wouldn't add minor version in the boot entry, I don't think boot entry title should change after installing ordinary updates. So make it `Qubes (R4.2)` or `Qubes R4.2`.

Comment (alimirjamali) [reply grub/grub.qubes-kernel-vm-support]: Ok. I made it like this:
```
GRUB_DISTRIBUTOR="sed 's,\(.*\) release \(.*\) (R\(.*\)),\1 (R\2),g'" 
```

The 1st regex group catpures `Qubes`, the 2nd group captures `4.2.2` and the 3rd group captures `4.2`

The output is `Qubes (R4.2.2)`

It includes the minor version which is handy for debugging, troubleshooting and support matters. This is how I thought should be the best case. Please let me know if this fine or should be changed.

Comment (alimirjamali) [reply grub/grub.qubes-kernel-vm-support]: > I wouldn't add minor version in the boot entry, I don't think boot entry title should change after installing ordinary updates. So make it `Qubes (R4.2)` or `Qubes R4.2`.

Oh. I just missed this. Will do it right now

**Comment by marmarek** [comment]
> I don't think so. The `error: GPGME error: Inappropriate ioctl for device` error is something else and I think this is what later is seen as package signature verification failure.

https://github.com/QubesOS/qubes-builderv2/pull/137

**Comment by alimirjamali** [comment]
Ouch. I forgot the add `$()` to the sed command.

p.s. Fixed

**Comment by marmarek** [comment]
PipelineRetry

**Review by marmarek — CHANGES_REQUESTED** [review]
This config applies to a VMs too. And if fails there on Debian:
```
debian-12-xfce:out: update-initramfs: Generating /boot/initrd.img-6.1.0-23-amd64
debian-12-xfce:out: /etc/kernel/postinst.d/zz-update-grub:
debian-12-xfce:out: sed: can't read /etc/system-release: No such file or directory
debian-12-xfce:out: run-parts: /etc/kernel/postinst.d/zz-update-grub exited with return code 2
```
Better make this part apply only to dom0, for example by checking for /etc/qubes-release presence

**Comment by alimirjamali** [comment]
Done

**Review by marmarek — APPROVED** [review]

**Pull request merged**